### PR TITLE
Allows passing an object with string property name for listeners, removeObjectListener for removing all listeners based on that object context

### PR DESCRIPTION
**Pull request by jrburke**
Allows these styles of .on() calls:
- `on(eventId, Function)` where undefined will be use as "this" context when Function is called.
- `on(eventId, Object, String)` where String is a property name on Object. Object[String] will be called with Object as the "this" context.
- `on(eventId, Object, Function)` where object will be use as "this" context when Function is called.

Also introduces `evt.removeObjectListener(obj)` that will remove all listeners obj has attached to that evt.Emitter instance.

### CODE REVIEW

File: test/object_listen_test.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    };
+
+    evt.once('roll', obj, 'roll');
+    evt.emit('roll');
+    evt.emit('roll');
+
+    assert.equal(1, obj.rollCount);
+  });
+
+  it('once, but removed in removeObjectListener', function() {
```
Comment (asutherland) [inline test/object_listen_test.js:58]: very well thought-through edge-case coverage with this test!

**Comment by asutherland** [comment]
@jrburke r=asuth
